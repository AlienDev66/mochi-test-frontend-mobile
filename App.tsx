import React from 'react';
import { ProfilesProvider } from './app/hooks/useProfiles';
import FindGithubUsersAndCompanies from './app/screen/findGithubUsersAndCompanies';

export default function App() {
    return (
        <ProfilesProvider>
            <FindGithubUsersAndCompanies />
        </ProfilesProvider>
    );
}
