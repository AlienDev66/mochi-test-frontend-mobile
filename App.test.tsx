import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('<App />', () => {
    const tree = renderer.create(<App />).toJSON();
    it('has 3 child', () => {
        if (tree) expect(tree.children.length).toBe(3);
    });
    it('has 3 child', () => {
        if (tree) expect(tree).toMatchSnapshot();
    });
});
