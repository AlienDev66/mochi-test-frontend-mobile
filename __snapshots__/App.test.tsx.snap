// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has 2 child 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": "35%",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flexDirection": "column",
            "fontSize": 35,
            "fontWeight": "bold",
            "marginBottom": 20,
            "paddingLeft": 25,
          },
        ]
      }
    >
      Search for
      

      Github Users
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#acb7c5",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "marginLeft": 25,
            "marginRight": 25,
            "width": 300,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Type a user name here"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "black",
              "height": 50,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "width": "85%",
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#c200b9",
            "borderRadius": 2,
            "height": 32,
            "justifyContent": "center",
            "marginRight": 5,
            "opacity": 1,
            "width": 32,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "marginTop": 50,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "color": "#192538",
              "fontSize": 17,
              "marginTop": 25,
              "textAlign": "center",
              "width": 240,
            },
          ]
        }
      >
        Enter a login, name or a company you are looking for.
      </Text>
    </View>
  </View>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#D1D9E2",
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    SHOW MORE
  </Text>
</View>
`;

exports[`<App /> has 3 child 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": "35%",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flexDirection": "column",
            "fontSize": 35,
            "fontWeight": "bold",
            "marginBottom": 20,
            "paddingLeft": 25,
          },
        ]
      }
    >
      Search for
      

      Github Users
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#acb7c5",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "marginLeft": 25,
            "marginRight": 25,
            "width": 300,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Type a user name here"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "black",
              "height": 50,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "width": "85%",
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#c200b9",
            "borderRadius": 2,
            "height": 32,
            "justifyContent": "center",
            "marginRight": 5,
            "opacity": 1,
            "width": 32,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "marginTop": 50,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "color": "#192538",
              "fontSize": 17,
              "marginTop": 25,
              "textAlign": "center",
              "width": 240,
            },
          ]
        }
      >
        Enter a login, name or a company you are looking for.
      </Text>
    </View>
  </View>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#D1D9E2",
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    SHOW MORE
  </Text>
</View>
`;
